libs/net/core/Extensions/StringExtensions.cs

```diff
@@ -303,21 +303,22 @@ public static string SanitizeContent(string articleContent, string tagName, stri
 
     private static string SanitizeContent(string articleContent, Regex regex)
     {
+        var result = articleContent;
         var matches = regex.Matches(articleContent);
         foreach (var match in matches.Cast<Match>())
         {
             var index = match.Index + match.Value.Length;
             if (articleContent.Length > index && articleContent[index] != '.')
             {
-                articleContent = articleContent.Replace(
+                result = result.Replace(
```

Comment:
Can you add a code comment that explains what all this is doing.  It's difficult to read and understand.

Reply:
Comment added, and hope this helps.